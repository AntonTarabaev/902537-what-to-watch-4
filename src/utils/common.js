import {MINUTES_IN_HOUR, MONTHS} from "@constants/main";

export const formatTime = (durationInMinutes) => {
  return (
    durationInMinutes > MINUTES_IN_HOUR ?
      `${(durationInMinutes - durationInMinutes % MINUTES_IN_HOUR) / MINUTES_IN_HOUR}h ${durationInMinutes % MINUTES_IN_HOUR}m` :
      `${durationInMinutes}m`
  );
};

export const formatDate = (date) => {
  return `${MONTHS[date.getMonth()]} ${date.getDate()}, ${date.getFullYear()}`;
};
