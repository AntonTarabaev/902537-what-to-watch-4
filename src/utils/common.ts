import {MINUTES_IN_HOUR, MONTHS} from "@constants/main";
import {Film} from "@root/types";

export const formatTime = (durationInMinutes: number) => {
  return (
    durationInMinutes > MINUTES_IN_HOUR ?
      `${(durationInMinutes - durationInMinutes % MINUTES_IN_HOUR) / MINUTES_IN_HOUR}h ${durationInMinutes % MINUTES_IN_HOUR}m` :
      `${durationInMinutes}m`
  );
};

export const formatDate = (date: Date) => {
  return `${MONTHS[date.getMonth()]} ${date.getDate()}, ${date.getFullYear()}`;
};

export const extend = (a: object, b: object) => {
  return Object.assign({}, a, b);
};

export const getUniqueGenres = (films: Film[], maxGenresCount: number) => {
  return [...new Set(films.map((it) => it.genre))].slice(0, maxGenresCount);
};

export const changeFormElementsDisabledProperty = (form: HTMLFormElement, disabled: boolean) => {
  Array.from(form.elements).map((formElement: HTMLInputElement) => {
    formElement.disabled = disabled;
  });
};

export const noop = () => {
  // do nothing
};
