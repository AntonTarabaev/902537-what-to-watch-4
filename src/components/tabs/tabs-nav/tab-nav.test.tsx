import TabNav from "@components/tabs/tabs-nav/tab-nav";
import * as renderer from "react-test-renderer";
import {noop} from "@utils/common";

const TabInfo = {
  title: `Overview`,
  number: 0,
};

describe(`TabNav component render correctly`, () => {
  it(`with active tab`, () => {
    const tree = renderer.create(
        <TabNav
          title={TabInfo.title}
          number={TabInfo.number}
          activeTabId={TabInfo.number}
          tabTitleClickHandler={noop}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`with not active tab`, () => {
    const tree = renderer.create(
        <TabNav
          title={TabInfo.title}
          number={TabInfo.number}
          activeTabId={TabInfo.number + 1}
          tabTitleClickHandler={noop}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
