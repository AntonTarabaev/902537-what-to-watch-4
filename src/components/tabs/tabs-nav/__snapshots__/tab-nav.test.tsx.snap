// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabNav component render correctly with active tab 1`] = `
<li
  className="movie-nav__item movie-nav__item--active"
>
  <a
    className="movie-nav__link"
    href={null}
    onClick={[Function]}
  >
    Overview
  </a>
</li>
`;

exports[`TabNav component render correctly with not active tab 1`] = `
<li
  className="movie-nav__item "
>
  <a
    className="movie-nav__link"
    href="#"
    onClick={[Function]}
  >
    Overview
  </a>
</li>
`;
